core/src/test/java/org/apache/iceberg/rest/TestHTTPClient.java

```diff
@@ -121,6 +128,108 @@ public void testHeadFailure() throws JsonProcessingException {
     testHttpMethodOnFailure(HttpMethod.HEAD);
   }
 
+  @Test
+  public void testProxyServer() throws IOException {
+    int proxyPort = 1070;
+    try (ClientAndServer proxyServer = startClientAndServer(proxyPort);
+        RESTClient clientWithProxy =
+            HTTPClient.builder(ImmutableMap.of())
+                .uri(URI)
+                .withProxy("localhost", proxyPort)
+                .build()) {
+      String path = "v1/config";
+
+      HttpRequest mockRequest =
+          request("/" + path).withMethod(HttpMethod.HEAD.name().toUpperCase(Locale.ROOT));
+
+      HttpResponse mockResponse = response().withStatusCode(200);
+
+      mockServer.when(mockRequest).respond(mockResponse);
+      proxyServer.when(mockRequest).respond(mockResponse);
+
+      clientWithProxy.head(path, ImmutableMap.of(), (onError) -> {});
+      proxyServer.verify(mockRequest, VerificationTimes.exactly(1));
+    }
+  }
+
+  @Test
+  public void testProxyCredentialProviderWithoutProxyServer() {
```

Comment:
I think the entire test can be simplified to 
```
@Test
  public void testProxyCredentialProviderWithoutProxyServer() {
    Assertions.assertThatThrownBy(
            () -> HTTPClient.builder(ImmutableMap.of())
                .uri(URI)
                .withProxyCredentialsProvider(new BasicCredentialsProvider())
                .build())
        .isInstanceOf(NullPointerException.class)
        .hasMessage("Invalid http client proxy for proxy credentials provider: null");
  }
```
because it's a bit confusing if the test description mentions `WithoutProxyServer` but the code actually configures `proxy`